File: week-2/most-common-word/uncommon-words-from-two-sentences.js
/**
 * @param {string} A
 * @param {string} B
 * @return {string[]}
 */
var uncommonFromSentences = function(A, B) {
    const hashtable = {}
    for (let i of A.split(' ').concat(B.split(' '))) {
        hashtable[i] = i in hashtable ? ++hashtable[i] : 1;
    }
    
    return Object.entries(hashtable).filter(item => item[1] === 1).map(item => item[0]);
};

File: week-2/most-common-word/uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, A: str, B: str) -> List[str]:
        hashtable = {}
        for i in A.split() + B.split():
            hashtable[i] = hashtable[i] + 1 if i in hashtable else 1
            
        return [key for key, val in hashtable.items() if val == 1]
